controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -184,14 +184,14 @@ private boolean changesAvailable(Application application, JobStatus previous, Jo
         if ( ! application.deploying().isPresent()) return false;
         Change change = application.deploying().get();
 
-        if ( ! previous.lastSuccess().isPresent() && 
+        if ( ! previous.isSuccess() &&
              ! productionUpgradeHasSucceededFor(previous, change)) return false;
 
         if (change instanceof Change.VersionChange) {
             Version targetVersion = ((Change.VersionChange)change).version();
             if ( ! (targetVersion.equals(previous.lastSuccess().get().version())) )
                 return false; // version is outdated
-            if (isOnNewerVersionInProductionThan(targetVersion, application, next.type()))
+            if (next != null && isOnNewerVersionInProductionThan(targetVersion, application, next.type()))
```

Comment:
next is null when that job type hasn't been run before.

Reply:
Thanks!